File: java/src/com/github/liupu14/DayOfProgrammer.java
package com.github.liupu14;

import java.util.Scanner;

public class DayOfProgrammer {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int year = in.nextInt();
        printDayOfProgrammer(year);
    }

    private static void printDayOfProgrammer(int year) {
        if (year == 1918){
            System.out.println("26.09.1918");
        } else if (year <= 1917){
            if (year % 4 == 0){
                System.out.println("12.09."+year);
            } else{
                System.out.println("13.09."+year);
            }
        } else {
            if ((year % 400 == 0) || (year % 4 == 0 && year % 100 != 0)){
                System.out.println("12.09."+year);
            } else {
                System.out.println("13.09."+year);
            }
        }
    }
}

File: python/dayOfProgrammer.py
def print_day_of_programmer(year):
    if year == 1918:
        print("26.09.1918")
    elif year <= 1917:
        if year % 4 == 0:
            print("12.09.{}".format(year))
        else:
            print("13.09.{}".format(year))
    else:
        if year % 400 == 0 or (year % 4 == 0 and year % 100 != 0):
            print("12.09.{}".format(year))
        else:
            print("13.09.{}".format(year))


if __name__ == "__main__":
    year = int(input("请输入年份: "))
    print_day_of_programmer(year)
